test/e2e/resize_nodes.go

```diff
@@ -421,6 +421,14 @@ var _ = Describe("Nodes", func() {
 			if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes, 10*time.Minute); err != nil {
 				Failf("Couldn't restore the original cluster size: %v", err)
 			}
+			// Many e2e tests assume that the cluster is fully healthy before they start.  Wait until
+			// the cluster is restored to health
+			By("waiting for system pods to successfully restart")
+			pods, err := framework.Client.Pods("kube-system").List(labels.Everything(), fields.Everything())
```

Comment:
Wouldn't it introduce some races?
It may happen that the node is ready, but kube-system pods are not yet created - or am I missing something? If that's not possible, can you please add a comment?